tests/lightning_checkpoint_engines/run.sh

```diff
@@ -81,13 +81,17 @@ for BACKEND in importer local; do
   run_sql 'DROP DATABASE IF EXISTS tidb_lightning_checkpoint;'
   rm -rf $TEST_DIR/lightning_checkpoint_engines.sorted
 
-  set +e
   for i in $(seq "$ENGINE_COUNT"); do
       echo "******** Importing Table Now (step $i/$ENGINE_COUNT) ********"
+      set +e
       do_run_lightning $BACKEND mysql 2> /dev/null
       [ $? -ne 0 ] || exit 1
+      set -e
+      # make sure the data engines are processed in the engine_id order.
+      # there is 75% this test failing if they are processed randomly.
+      run_sql 'SELECT *, status <= prev_engine_status AS ordering FROM (SELECT table_name, engine_id, status, lag(status) OVER (PARTITION BY table_name ORDER BY engine_id) AS prev_engine_status FROM tidb_lightning_checkpoint.engine_v5 WHERE engine_id >= 0) _'
```

Comment:
So I think this check should be `status < prev_engine_status` or add another restriction that `status >= 120`

Reply:
changed to a more dedicated test case.